xml/operating_system_darwin.xml

```diff
@@ -0,0 +1,110 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  Patterns for common names of Mac OS X operating systems.
+-->
+<fingerprints matches="operating_system.name">
+  <fingerprint pattern="^Mac OS X\s?(\d+?(?:\.\d+?)+?)?$" flags="REG_ICASE">
```

Comment:
You're right, I didn't consider only providing the major version. Simple enough to use a `*` instead of `+` on the repetition of the nested versions.